specs/workflow-clone-spec.yaml

```diff
@@ -0,0 +1,129 @@
+        '200':
+          description: Successfully cloned the application workflow
+          content:
+            application/json:
+              schema:
+                $ref: '#/components/schemas/CloneApplicationWorkflowResponse'
+        '400':
+          description: Bad Request. Input validation error or wrong request body.
+          content:
+            application/json:
+              schema:
+                $ref: '#/components/schemas/DetailedError'
+        '500':
+          description: Internal Server Error
+          content:
+            application/json:
+              schema:
+                $ref: '#/components/schemas/DetailedError'
+        '401':
+          description: Unauthorized User
+          content:
+            application/json:
+              schema:
+                $ref: '#/components/schemas/DetailedError'
+        '403':
+          description: Forbidden. User does not have the necessary permissions.
+          content:
+            application/json:
+              schema:
+                $ref: '#/components/schemas/DetailedError'
+
+components:
+  schemas:
+    CloneApplicationWorkflowRequest:
+      type: object
+      properties:
+        appId:
+          type: integer
+          description: ID of the application
+        sourceEnvironmentId:
+          type: integer
+          description: ID of the source environment
+        targetEnvironmentId:
+          type: integer
+          description: ID of the target environment
+        cloneEnvInSameWorkflow:
+          type: boolean
+          description: Flag indicating if the environment should be cloned in the same workflow
+      required:
+        - appId
+        - sourceEnvironmentId
+        - targetEnvironmentId
+        - cloneEnvInSameWorkflow
+    CloneApplicationWorkflowResponse:
+      type: object
+      properties:
+        code:
+          type: integer
+          description: Response code
+        status:
+          type: string
+          description: Status message
+        result:
+          type: object
+          properties:
+            appId:
+              type: integer
+              description: ID of the application
+            sourceEnvironmentId:
+              type: integer
+              description: ID of the source environment
+            targetEnvironmentId:
+              type: integer
+              description: ID of the target environment
+            cloneEnvInSameWorkflow:
+              type: boolean
+              description: Flag indicating if the environment was cloned in the same workflow
+      required:
+        - code
+        - status
+        - result
+    DetailedError:
+      type: object
+      properties:
+        code:
+          type: integer
+          description: HTTP status code
+        status:
+          type: string
+          description: Status message
+        errors:
+          type: array
+          items:
+            $ref: '#/components/schemas/ErrorDetail'
+    ErrorDetail:
```

Comment:
We should not throw 400, 500 in case of Skipped and Failed since they are not actually errors, rather can receive a enum to determine Skipped and Failed with status code 200 and message key to infer message so that code like 500 is actually reserved for Internal Server Error.